analyses/tmb-vaf-longitudinal/03-explore-TMB-mut.Rmd

```diff
@@ -0,0 +1,221 @@
+ylim = max(df$tmb)
+
+# Re-order df
+f <- c("Second Malignancy", "Unavailable", "Deceased", "Recurrence", "Progressive", "Diagnosis") # Level df by timepoints
+df_plot <- df %>% 
+  dplyr:::mutate(tumor_descriptor = factor(tumor_descriptor),
+                 tumor_descriptor = fct_relevel(tumor_descriptor, f)) 
+
+# Run function
+fname <- paste0(plots_dir, "/", "TMB-genomic-total.pdf")
+print(fname)
+p <- create_stacked_barplot(tmb_df = df_plot, ylim = ylim)
+pdf(file = fname, width = 15, height = 6)
+print(p)
+dev.off()
+```
+Attention: Hypermutant TMB defined as ≥10 Mb, and Ultrahypermutant TMB defined as ≥100 mutations/Mb in pediatric brain tumors (https://pubmed.ncbi.nlm.nih.gov/29056344/).
+
+Here, we notice that there are samples with high TMB (hyper-mutant samples). Next, we will exclude these samples (threshold >= 10) from downstream analysis. Attention is needed in cases with high number of mutations in only one timepoint as this will lead to un-matched longitudinal samples. We will also remove those so we always have matched longitudinal samples.
+
+```{r create-stacked-barplot-filter, fig.width = 15, fig.height = 8, fig.fullwidth = TRUE}
+# Filter df and remove any samples with single timepoints
+df_plot_filter <- df %>%
+  filter(!tmb >= 10) %>%
+  unique() %>% 
+  arrange(Kids_First_Participant_ID, tumor_descriptor) %>%
+  group_by(Kids_First_Participant_ID) %>%
+  dplyr::summarise(tumor_descriptor_sum = str_c(tumor_descriptor, collapse = ";")) %>% 
+  filter(!tumor_descriptor_sum %in% c("Diagnosis", "Progressive", "Recurrence", "Second Malignancy", "Unavailable", "Deceased", "Progressive;Progressive")) %>% 
+  dplyr::left_join(df, by = c("Kids_First_Participant_ID", "tumor_descriptor_sum")) %>% 
+  mutate(tumor_descriptor = factor(tumor_descriptor),
+         tumor_descriptor = fct_relevel(tumor_descriptor, f)) %>% 
+  drop_na(tmb) 
+
+# length(unique(df_plot_filter$Kids_First_Participant_ID))
+
+# Define parameters for function
+ylim = max(df_plot_filter$tmb)
+df_plot_filter <- df_plot_filter
+
+# Run function
+fname <- paste0(plots_dir, "/", "TMB-genomic-no-hypermutants.pdf")
+print(fname)
+p <- create_stacked_barplot(tmb_df = df_plot_filter, ylim = ylim)
+pdf(file = fname, width = 25, height = 8)
+print(p)
+dev.off()
+```
+
+# TMB across timepoints and cancer types per Patient case
+We will explore TMB per cancer group over time by creating dumbbell plots. We classified by using cancer types with the highest number of samples (High- and Low-grade gliomas) versus any other cancer groups.
+
+```{r create-dumbbell-ct, fig.width = 12, fig.height = 8, fig.fullwidth = TRUE}
+# How many bs_samples per kids_id and cancer group?
+# print(table(df_plot_filter$cg_plot))
+print(df_plot_filter %>% 
+        count(cg_plot, Kids_First_Participant_ID))
+
+# Dumbbell plot per cancer group
+cancer_groups <- unique(as.character(df_plot_filter$cg_plot))
+cancer_groups <- sort(cancer_groups, decreasing = FALSE)
+print(cancer_groups)
```

Comment:
This is a great question. I am addressing this issue but identifying samples with different diagnosis across timepoints and summarizing them in the `cg_sum` col generated in the `read_input_files` code chunk. That means if there are cases with different diagnosis at timepoints, there will show as, e.g., `Low-grade glioma,High-grade glioma`. If it is the same diagnosis across timepoints it will show as e.g., `Low-grade glioma`. Plots are named accordingly.

Parts of the relevant code showing below:

```
%>% mutate(cg_distinct = n_distinct(cancer_group) > 1) # to identify samples with different diagnosis across timepoints
```
<more_code>

```
%>% summarise(cg_sum = str_c(cancer_group, collapse = ",")) %>% # to identify cases with multiple diagnosis
```

Here, I am using the `cg_plot` col in which I have summarized the samples with less than 10 bs as other and retaining the information from the `cg_sum` column. 


```
# Let's summarize cancer groups with < 10 bs_samples as Other and use this for visualization purposes
cg_sum_df <- df %>% 
  count(cg_sum) %>% 
  dplyr::mutate(cg_sum_n = glue::glue("{cg_sum} (N={n})"))

df <- df %>% 
  left_join(cg_sum_df, by = c("cg_sum")) %>% 
  mutate(cg_plot = case_when(n < 10 ~ "Other",
                             TRUE ~ cg_sum))

```


@rjcorb Please let me know if more clarification is needed.